src/sentry/middleware/env.py

```diff
@@ -14,3 +16,12 @@ def process_request(self, request):
 
         # bind request to env
         env.request = request
+
+        # Periodically for an expire the local OptionsStore cache.
+        # This cleanup is purely to keep memory low and garbage collect
+        # old values. It's not required to run to keep things consistent.
+        # Internally, if an option is fetched and it's expired, it gets
+        # evicted immediately. This is purely for options that haven't
+        # been fetched since they've expired.
+        if random() < 0.25:
```

Comment:
I'm open to just doing this on every request, but figured it was unnecessary since it's a best effort kinda thing.

Feel free to push back if you have a strong opinion.